infoblox/resourceIPAllocation.go

```diff
@@ -62,32 +80,47 @@ func resourceIPAllocation() *schema.Resource {
 	}
 }
 
-//We are using creation of host record for IPAM purposes for the reason that
-//if we create a fixed address,t has mac address as a required parameter and
-//a mac address is not generated before provisioning of vm hence we use create
-//host record method used below
 func resourceIPAllocationRequest(d *schema.ResourceData, m interface{}) error {
 	log.Printf("[DEBUG] %s: Beginning to request a next free IP from a required network block", resourceIPAllocationIDString(d))
 
 	networkViewName := d.Get("network_view_name").(string)
-	recordName := d.Get("host_name").(string)
+	//This is for record Name
+	record_Name := d.Get("vm_name").(string)
 	ipAddr := d.Get("ip_addr").(string)
 	cidr := d.Get("cidr").(string)
 	macAddr := d.Get("mac_addr").(string)
 	vmID := d.Get("vm_id").(string)
+	//This is for EA's
+	vmName := d.Get("vm_name").(string)
 	tenantID := d.Get("tenant_id").(string)
+	zone := d.Get("zone").(string)
+	enableDns := d.Get("enable_dns").(bool)
+	dnsView := d.Get("dns_view").(string)
+
 	connector := m.(*ibclient.Connector)
+	ZERO_MACADDR := "00:00:00:00:00:00"
+	name := record_Name + "." + zone
 
+	if macAddr == "" {
+		macAddr = ZERO_MACADDR
+	}
 	objMgr := ibclient.NewObjectManager(connector, "terraform", tenantID)
 
-	hostrecordObj, err := objMgr.CreateHostRecordWithoutDNS(recordName, networkViewName, cidr, ipAddr, macAddr, vmID)
-	if err != nil {
-		return fmt.Errorf("Error allocating IP from network block(%s): %s", cidr, err)
+	if (zone != "" || len(zone) != 0) && (dnsView != "" || len(dnsView) != 0) {
```

Comment:
Both check are same right. why we need this 2 check. 

zone != ""  == len(zone) != 0

Reply:
zone != "" == len(zone) != 0  cant be done because
invalid operation: dnsView != "" == len(dnsView) (mismatched types bool and int)
i've see the core NIOS code have these kind of check that is why i added them